sympy/solvers/ode/tests/test_ode.py

```diff
@@ -160,18 +160,8 @@ def test_linear_2eq_order1_type2_fixme_check():
     assert checksysodesol(eqs, sol) == (True, [0, 0])
 
 
-def test_linear_2eq_order1_type4():
-    eqs = [Eq(diff(f(x), x),  f(x) + x*g(x)),
-           Eq(diff(g(x), x),-x*f(x) + g(x))]
-    sol = [Eq(f(x), (C1*cos(x**2/2) + C2*sin(x**2/2))*exp(x)),
-           Eq(g(x), (-C1*sin(x**2/2) + C2*cos(x**2/2))*exp(x))]
-    dsolve_sol = dsolve(eqs)
-    # FIXME: This should probably be fixed so that this happens in the solver:
-    dsolve_sol = [s.doit() for s in sol]
-    assert dsolve_sol == sol
-    assert checksysodesol(eqs, sol) == (True, [0, 0])
-
-
+# This test is not fixed by the new solver because the
+# coefficient matrix's antiderivative doesn't commute.
 def test_linear_2eq_order1_type4_broken():
```

Comment:
@oscarbenjamin I think this is a faulty test case. This test case doesn't have a commutative antiderivative and doesn't even have the form described in Linear, 2 equations, Order 1, Type 4. Should I remove this test case altogether?

Reply:
There can be solutions to systems that don't have a commutative antiderivative so that is not a reason for removing a test case if it is currently solved by one of the solvers.

However in this case it looks as if this example should not match any of the existing solvers and the solution given by type4 is incorrect.

Yes I think it's fine to remove this.

If we knew the correct answer for the system of ODEs then it would be good to add that as an XFAIL test instead though.

Reply:
I can't find what the solution should be in this case. It's possible that there is no solution in ordinary functions. I think it's best to remove this example.

Reply:
> There can be solutions to systems that don't have a commutative antiderivative so that is not a reason for removing a test case if it is currently solved by one of the solvers.

Ok, I will keep this in mind from now on when dealing with XFail tests.

>  I think it's best to remove this example.

Sure thing, I will remove this example.